go-chromecast: fix build by using `buildGo122Module`

zi3m5f opened the pull request:
ZHF: #352882

Fails on `0.3.1` and `upstream/master` with:
> Running phase: checkPhase
> github.com/rogpeppe/go-internal/testscript
vendor/github.com/rogpeppe/go-internal/testscript/exe_go118.go:14:27: cannot use nopTestDeps{} (value of type nopTestDeps) as testing.testDeps value in argument to testing.MainStart: nopTestDeps does not implement testing.testDeps (missing method InitRuntimeCoverage)
FAIL    github.com/vishen/go-chromecast [build failed]

Probably caused by changes in go 1.23.

Other changes while at it:
- adopt (no hardware to test)
- add `versionCheckHook`
- add shell completions with `installShellFiles`
- nixfmt
- move to `pkgs/by-name`
- other improvements

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

zi3m5f committed:
go-chromecast: nixfmt

Aleksanaa commented on pkgs/applications/video/go-chromecast/default.nix:
```diff
@@ -1,24 +1,44 @@
+  ];
 
-  meta = with lib; {
+  doCheck = false; # fails to build
```
```suggestion
  doCheck = false; # fails to build tests
```

Aleksanaa reviewed (approved):
Would be good to split other modifications to another commit

fabaff commented on pkgs/applications/video/go-chromecast/default.nix:
```diff
@@ -1,24 +1,44 @@
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+    "-X main.commit=${src.rev}"
+    "-X main.date=unknown"
```
```suggestion
    "-X=main.version=${version}"
    "-X=main.commit=${src.rev}"
    "-X=main.date=unknown"
```

Aleksanaa commented on pkgs/by-name/go/go-chromecast/package.nix:
```diff
@@ -0,0 +1,58 @@
+    homepage = "https://github.com/vishen/go-chromecast";
+    description = "CLI for Google Chromecast, Home devices and Cast Groups";
+    license = lib.licenses.asl20;
+    maintainers = [ ];
```
Would you like to maintain this package?

fabaff commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353037`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>go-chromecast</li>
  </ul>
</details>

Aleksanaa commented:
> I found out an alternative fix would be to just use go 1.22 with `buildGo122Module` instead. Would that be better? Sorry, no go expert ;)

A little better, I think. But we still hope that upstream will fix it eventually, because Go versions are updated very quickly and we will not provide support for deprecated versions.

zi3m5f committed:
go-chromecast: fix build

"go test" fails with go 1.23 use buildGo122Module for now.

zi3m5f committed:
go-chromecast: adopt and other improvements

- adopt (no hardware to test)
- add `versionCheckHook`
- add shell completions with `installShellFiles`
- other improvements

zi3m5f committed:
go-chromecast: move to pkgs/by-name
